cmake/version.cmake

```diff
@@ -23,5 +23,10 @@ else()
   set(REVISION_LAST "${REVISION}")
 endif()
 
+if (REVISION MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)")
+    set(REVISION_MAJOR ${CMAKE_MATCH_1})
+    set(REVISION_MINOR ${CMAKE_MATCH_2})
+    set(REVISION_PATCH ${CMAKE_MATCH_3})
+endif()
```

Comment:
It is probably worth handling the case when the match does not happen. What do you think is the best?
Likely something along:
```
message(FATAL_ERROR "Could extract major/minor/patch from revision ${REVISION}" )
```

would be the way to go.

Reply:
Another thing is that pulling tags directly from git may not work reliably when using another repo than ip visor/bcc. Like https://github.com/chantra/gobpf/runs/7140089907?check_suite_focus=true#step:4:215 when I used my own to test the macro detection. Proper messaging may be useful.

Reply:
Okay. But instead of FATAL_ERROR, could we add a warning. For official iovisor/bcc repo, we should be always okay. But for bcc repo forks with custom changes, a warning is enough so it won't break custom build. the custom build may not care about versions at all.